### PR TITLE
Add SQLSRV Integration

### PR DESCRIPTION
**Pull request by PROFeNoM**
### Description

This PR adds a SQLSRV Integration. Moreover, it will change the buster dockerfiles to include the required drivers (these images have already been pushed to datadog/dd-trace-ci)

### Readiness checklist
- [X] (only for Members) Changelog has been added to the release document.
- [X] Tests added for this feature/bug.

### Reviewer checklist
- [X] Appropriate labels assigned.
- [X] Milestone is set.
- [X] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: src/Integrations/Integrations/SQLSRV/SQLSRVIntegration.php [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                DatabaseIntegrationHelper::injectDatabaseIntegrationData($hook, 'sqlsrv', 1);
+            }, function (HookData $hook) use ($integration) {
+                $span = $hook->span();
+                if (is_object($hook->returned)) {
+                    ObjectKVStore::propagate($this, $hook->returned, SQLSRVIntegration::CONNECTION_TAGS_KEY);
+                }
+
+                $integration->detectError($hook->returned, $span);
+            });
+        } else {
```
Comment (pierotibou) [inline src/Integrations/Integrations/SQLSRV/SQLSRVIntegration.php:95]: In theory, as we're only in [maintenance mode for PHP5](https://docs.datadoghq.com/tracing/trace_collection/compatibility/php/#runtime-support-policy-for-php-apm), you didn't have to do this. But no worries if it is done now

**Review by pierotibou — APPROVED** [review]
LGTM, thanks

File: src/Integrations/Integrations/SQLSRV/SQLSRVIntegration.php (hunk 1 above)
Comment (bwoebi) [reply src/Integrations/Integrations/SQLSRV/SQLSRVIntegration.php:95]: (It just has no effect on PHP 5 as it's not declared in the IntegrationsLoader for PHP 5)

File: dockerfiles/ci/buster/php-7.0/Dockerfile [hunk 2]
```diff
@@ -50,6 +50,8 @@ COPY --chown=circleci:circleci --from=php-debug-zts $PHP_INSTALL_DIR/debug-zts $
 COPY --chown=circleci:circleci --from=php-debug $PHP_INSTALL_DIR/debug $PHP_INSTALL_DIR/debug
 COPY --chown=circleci:circleci --from=php-nts $PHP_INSTALL_DIR/nts $PHP_INSTALL_DIR/nts
 
+USER root
```
Comment (bwoebi) [inline dockerfiles/ci/buster/php-7.0/Dockerfile:53]: What's up with this? We're anyway using sudo everywhere in this dockerfile, where we need root?

File: ext/integrations/integrations.c [hunk 3]
```diff
@@ -223,8 +223,11 @@ void ddtrace_integrations_minit(void) {
     DD_SET_UP_DEFERRED_LOADING_BY_METHOD(DDTRACE_INTEGRATION_SYMFONY, "Symfony\\Component\\HttpKernel\\HttpKernel", "__construct",
                                          "DDTrace\\Integrations\\Symfony\\SymfonyIntegration");
 
+    DD_SET_UP_DEFERRED_LOADING_BY_FUNCTION(DDTRACE_INTEGRATION_SQLSRV, "sqlsrv_connect",
+                                         "DDTrace\\Integrations\\SQLSRV\\SQLSRVIntegration");
+
     DD_SET_UP_DEFERRED_LOADING_BY_FUNCTION(DDTRACE_INTEGRATION_WORDPRESS, "wp_check_php_mysql_versions",
-                                           "DDTrace\\Integrations\\WordPress\\WordPressIntegration");
+                                         "DDTrace\\Integrations\\WordPress\\WordPressIntegration");
```
Comment (bwoebi) [inline ext/integrations/integrations.c:230]: It's aligned with the (, not with the offset where the other lines have their " :-D

File: src/Integrations/Integrations/SQLSRV/SQLSRVIntegration.php [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+            );
+        }, $errors));
+
+        $span->meta[Tag::ERROR_MSG] = $errorMessages;
+        $span->meta[Tag::ERROR_TYPE] = 'SQLSRV error';
+    }
+
+    protected function setMetricNumRows(SpanData $span, $stmt)
+    {
+        if ($stmt) {
```
Comment (bwoebi) [inline src/Integrations/Integrations/SQLSRV/SQLSRVIntegration.php:245]: If there's no num_rows, can you please assign sqlsrv_rows_affected() to db.row_count?

**Review by bwoebi — APPROVED** [review]
Generally looks good to me.

I'm just a bit confused where the logic is to not build sqlsrv on arm64 for older versions? I thought we found that the sqlsrv extension does not work on older PHP versions with arm64, or do I misremember?

**Pull request merged**